types/time.go

```diff
@@ -2577,3 +2578,40 @@ func DateFSP(date string) (fsp int) {
 	}
 	return
 }
+
+var (
+	dateRange = []gotime.Time{
+		gotime.Date(1000, 1, 1, 0, 0, 0, 0, gotime.Local),
+		gotime.Date(9999, 12, 31, 0, 0, 0, 0, gotime.Local),
+	}
+	timestampRange = []gotime.Time{
+		gotime.Date(1970, 1, 1, 0, 0, 1, 0, gotime.Local),
+		gotime.Date(2038, 1, 19, 3, 14, 7, 0, gotime.Local),
+	}
+	datetimeRange = []gotime.Time{
+		gotime.Date(1000, 1, 1, 0, 0, 0, 0, gotime.Local),
+		gotime.Date(9999, 12, 31, 23, 59, 59, 0, gotime.Local),
+	}
+)
+
+// DateTimeIsOverflow return if this date is overflow.
+// See: https://dev.mysql.com/doc/refman/8.0/en/datetime.html
+func DateTimeIsOverflow(date Time) (bool, error) {
+	var b, e gotime.Time
+	switch date.Type {
+	case mysql.TypeDate:
+		b, e = dateRange[0], dateRange[1]
+	case mysql.TypeTimestamp:
+		b, e = timestampRange[0], timestampRange[1]
```

Comment:
should we consider the system timezone info for `timestamp` values?

Reply:
You are right.
I will fix this.

Reply:
Updated.